pkg/apis/monitoring/v1/types.go

```diff
@@ -102,6 +102,11 @@ type ProxyConfig struct {
 	// +optional
 	// +mapType:=atomic
 	ProxyConnectHeader map[string][]v1.SecretKeySelector `json:"proxyConnectHeader,omitempty"`
+
+	// It requires Prometheus >= v2.55.0.
+	//
+	// +optional
+	HTTPHeadersConfig `json:",inline"`
```

Comment:
IIUC the field is added to HTTP configuration, not Proxy configuration?

Reply:
There is currently no global HTTPConfig, add a new HTTPConfig struct as name `HTTPConfig`
```
type HTTPConfig struct {
        // It requires Prometheus >= v2.55.0.
	//
	// +optional
	HTTPHeadersConfig `json:",inline"`
}
```
@simonpasquier WDYT?

Reply:
Hmm I think that the `inline` stanza isn't at the right place?

````

type Endpoint struct {
    ...

    HTTPConfig `json:",inline"`
}

type HTTPConfig struct {
        // It requires Prometheus >= v2.55.0.
	//
	// +optional
	HTTPHeaders map[string]HTTPHeader `json:"httpHeaders,omitempty"`
}
````

Over time we could move all the common HTTP fields to the new HTTPConfig (it will be tedious but it doesn't have to happen all at once).

Reply:
Thanks. I got it.
Change `HTTPConfig` to `CustomHTTPConfig` because it conflicts with the `HTTPConfig` struct with the same name in `alertmanager_types.go`